jerry-core/ecma/base/ecma-helpers-conversion.c

```diff
@@ -1074,204 +950,119 @@ ecma_number_to_utf8_string (ecma_number_t num, /**< ecma-number */
                             lit_utf8_byte_t *buffer_p, /**< buffer for utf-8 string */
                             lit_utf8_size_t buffer_size) /**< size of buffer */
 {
-  const lit_utf8_byte_t digits[10] = { '0', '1', '2', '3', '4', '5', '6', '7', '8', '9' };
-  const lit_utf8_byte_t e_chars[2] = { 'e', 'E' };
-  const lit_utf8_byte_t plus_char = '+';
-  const lit_utf8_byte_t minus_char = '-';
-  const lit_utf8_byte_t dot_char = '.';
-  lit_utf8_size_t size;
+  lit_utf8_byte_t *dst_p;
 
   if (ecma_number_is_nan (num))
   {
     // 1.
-    lit_copy_magic_string_to_buffer (LIT_MAGIC_STRING_NAN, buffer_p, buffer_size);
-    size = lit_get_magic_string_size (LIT_MAGIC_STRING_NAN);
+    dst_p = lit_copy_magic_string_to_buffer (LIT_MAGIC_STRING_NAN, buffer_p, buffer_size);
+    return (lit_utf8_size_t) (dst_p - buffer_p);
   }
-  else
+
+  if (ecma_number_is_zero (num))
```

Comment:
As far as I remember that there was a bug in the current implementation that -0 was not working. Is this working now?